tools/gradle-plugin/src/test/kotlin/io/github/composegears/valkyrie/gradle/Utils.kt

```diff
@@ -0,0 +1,106 @@
+import kotlin.io.path.createDirectories
+import kotlin.io.path.exists
+import kotlin.io.path.inputStream
+import kotlin.io.path.writeText
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.gradle.testkit.runner.TaskOutcome.SUCCESS
+
+internal fun buildRunner(root: Path, androidHome: Path? = null) = GradleRunner
+    .create()
+    .withPluginClasspath()
+    .withGradleVersion(System.getProperty("test.version.gradle"))
+    .withProjectDir(root.toFile())
+    .apply {
+        if (androidHome != null) {
+            withEnvironment(mapOf("ANDROID_HOME" to androidHome.absolutePathString()))
+        }
+    }
+
+internal fun runTask(root: Path, task: String) = buildRunner(root).runTask(task)
+
+internal fun GradleRunner.runTask(task: String) = withArguments(
+    task,
+    "--configuration-cache",
+    "--info",
+    "--stacktrace",
+    "-Pandroid.useAndroidX=true", // needed for android builds to work, unused otherwise
+)
+
+internal fun Path.writeSettingsFile() = resolve("settings.gradle.kts").writeText(
+    """
+        pluginManagement {
+            repositories {
+                mavenCentral()
+                google()
+                gradlePluginPortal()
+            }
+        }
+
+        dependencyResolutionManagement {
+            repositories {
+                mavenCentral()
+                google()
+            }
+        }
+    """.trimIndent(),
+)
+
+internal fun Path.writeTestSvgs(sourceSet: String) {
+    val destDir = resolve("src/$sourceSet/svg")
+    destDir.createDirectories()
+
+    val sourceDir = Paths.get(System.getProperty("test.dir.svg"))
+    sourceDir.copyToRecursively(destDir, followLinks = true)
+}
+
+internal fun Path.writeTestDrawables(sourceSet: String) {
+    val destDir = resolve("src/$sourceSet/res/drawable")
+    destDir.createDirectories()
+
+    val sourceDir = Paths.get(System.getProperty("test.dir.xml"))
+    sourceDir.copyToRecursively(destDir, followLinks = true)
+}
+
+internal fun Assert<BuildResult>.taskWasSuccessful(name: String) = taskHadResult(name, SUCCESS)
+
+internal fun Assert<BuildResult>.taskHadResult(name: String, expected: TaskOutcome) = given { result ->
+    val outcome = result.task(name)?.outcome
+    if (outcome == expected) return
+    fail(expected, outcome)
+}
+
+internal fun androidHome(): Path? {
+    val androidHomeProp = System.getProperty("test.androidHome")
+    val androidHome = Paths.get(androidHomeProp)
+    if (androidHomeProp.isNotBlank() && androidHome.exists()) {
+        return androidHome
+    }
+
+    val localProps = Paths.get(System.getProperty("test.localProps"))
+    if (localProps.exists()) {
+        val properties = Properties()
+        localProps.inputStream().use { properties.load(it) }
+        val sdkHome = Paths.get(properties.getProperty("sdk.dir"))
+        if (sdkHome.exists()) {
+            return sdkHome
+        }
+    }
+
+    return null
+}
```

Comment:
This logic should be written in the build script; we just need to provide `ANDROID_HOME`.

Reply:
Done